apps/webapp/app/hooks/useEnvironments.ts

```diff
@@ -1,23 +1,25 @@
-import { RouteMatch } from "@remix-run/react";
+import { UIMatch } from "@remix-run/react";
 import { MatchedProject, useOptionalProject } from "./useProject";
+import { AppData } from "~/utils/appData";
+import { Handle } from "~/utils/handle";
 
 export type ProjectJobEnvironment = MatchedProject["environments"][number];
 
-export function useEnvironments(matches?: RouteMatch[]) {
+export function useEnvironments<T = AppData>(matches?: UIMatch<T, Handle>[]) {
```

Comment:
This seems like it's losing type information here, can you explain what's going on with this type parameter?

Reply:
Since the UIMatch type expects 2 type parameters I had to pass something.
The 2nd parameter is required because we get type completions for the matches from that.
So I'll use (matches?: UIMatch<unknown, Handle>[]) in place of that type param.
Is that ok ?

Reply:
What's the return type of `useEnvironments` above? And can you run `pnpm run typecheck --filter webapp` and make sure there are no type errors.

Reply:
Hi @ericallam 

![typecheck](https://github.com/triggerdotdev/trigger.dev/assets/53973174/a51dc7c6-da36-487e-8c3f-00ec0bee0fe8)

The type check was successful.